cypress/utils/apps/query_enhancements/commands.js

```diff
@@ -31,3 +31,66 @@ Cypress.Commands.add('setQueryLanguage', (value) => {
     force: true,
   });
 });
+
+/**
+ * Creates a new data source connection with basic auth
+ * @param {Object} options Configuration options for the data source
+ * @param {string} options.name The name/title for the data source
+ * @param {string} options.url The endpoint URL for the data source
+ * @param {string} options.authType The authentication type (e.g. 'no_auth', 'basic_auth', etc.)
+ * @param {Object} [options.credentials] Optional credentials for auth types that require them
+ * @param {string} [options.credentials.username] Username for basic auth
+ * @param {string} [options.credentials.password] Password for basic auth
+ */
+Cypress.Commands.add('addDataSource', (options) => {
```

Comment:
whats specific about query enhancements ? this seems like it should be globally used